apps/archive/archive_media.py

```diff
@@ -37,33 +40,44 @@ def on_create(self, docs):
         for doc in docs:
             if 'media' not in doc or doc['media'] is None:
                 abort(400, description="No media found")
-
-            file, content_type, metadata = self.get_file_from_document(doc)
-            inserted = [doc['media']]
+            # check content type of video by python-magic
+            content_type = magic.from_buffer(doc['media'].read(1024), mime=True)
+            doc['media'].seek(0)
             file_type = content_type.split('/')[0]
-
-            self._set_metadata(doc)
-
-            try:
-                doc[ITEM_TYPE] = self.type_av.get(file_type)
-                doc[ITEM_STATE] = CONTENT_STATE.PROGRESS
-                rendition_spec = get_renditions_spec(no_custom_crops=True)
-                with timer('archive:renditions'):
-                    renditions = generate_renditions(file, doc['media'], inserted, file_type,
-                                                     content_type, rendition_spec, url_for_media)
-                doc['renditions'] = renditions
-                doc['mimetype'] = content_type
-                set_filemeta(doc, metadata)
-
-                add_activity('upload', 'uploaded media {{ name }}',
-                             'archive', item=doc,
-                             name=doc.get('headline', doc.get('mimetype')),
-                             renditions=doc.get('renditions'))
-            except Exception as io:
-                logger.exception(io)
-                for file_id in inserted:
-                    delete_file_on_error(doc, file_id)
-                abort(500)
+            if file_type == 'video' and app.config.get("VIDEO_SERVER_ENABLE"):
+                if not self.videoEditor.check_video_server():
+                    raise SuperdeskApiError(message="Cannot connect to videoserver", status_code=500)
+                # upload media to video server
+                self.upload_file_to_video_server(doc, content_type)
+                # get thumbnails for timeline bar
+                self.videoEditor.get_timeline_thumbnails(doc.get('media'), 40)
+            else:
```

Comment:
I think it should be structured a bit differently to avoid duplication, there are some common parts which are here and inside `upload_file_to_video_server`, imo it would be better to only do the upload there and create a method for that original upload handling with generating renditions, and then use the same code for both again.